src/it/java/org/eclipse/microprofile/starter/utils/WebpageTester.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2017-2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.starter.utils;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+public class WebpageTester {
+    private static final Logger LOGGER = Logger.getLogger(WebpageTester.class.getName());
+
+    /**
+     * Patiently try to wait for a web page and examine it
+     *
+     * @param url             address
+     * @param timeoutS        in seconds
+     * @param stringToLookFor string must be present on the page
+     */
+    public static void testWeb(String url, long timeoutS, String stringToLookFor) throws InterruptedException, IOException {
+        if (StringUtils.isBlank(url)) {
+            throw new IllegalArgumentException("url must not be empty");
+        }
+        if (timeoutS < 0) {
+            throw new IllegalArgumentException("timeoutS must be positive");
+        }
+        if (stringToLookFor == null ||
+                StringUtils.isBlank(stringToLookFor)) {
+            throw new IllegalArgumentException("stringToLookFor must contain a non-empty string");
+        }
+        String webPage = "";
+        long now = System.currentTimeMillis();
+        long startTime = now;
+        boolean found = false;
+        while (now - startTime < 1000 * timeoutS) {
```

Comment:
Any need for this while loop? Why not just open the connection and retrieve content?

Reply:
Will address in readme:  Some runtimes give you HTTP 404 or even HTTP 200 with empty response in the very moment the web app is being deployed. IMHO a race condition.  This solves it.  Our goal is to test Starter not to test particular runtimes....